spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala

```diff
@@ -83,9 +83,13 @@ class ClassNLLCriterion[@specialized(Float, Double) T: ClassTag]
       output = if (curTarget == -1) ev.zero
       else ev.times(ev.negative(input.valueAt(curTarget)), total_weight)
     } else if (input.dim() == 2) {
+      target.squeeze()
```

Comment:
Why there's a squeeze here? @zhichao-li We should not change the input data, right?